core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -152,8 +152,13 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
       sc.sparkUser,
       sc.appName,
       sc.conf,
-      sc.conf.getOption("spark.mesos.driver.webui.url").orElse(sc.ui.map(_.appUIAddress))
+      sc.conf.getOption("spark.mesos.driver.webui.url").orElse(sc.ui.map(_.appUIAddress)),
+      None,
+      None,
+      sc.conf.getOption("spark.mesos.driver.frameworkId")
     )
+
+    unsetFrameworkID(sc)
```

Comment:
If the driver creates two spark contexts shouldnt it use two different spark confs with that property set up differently?
If no id is specified what will happen? I guess the dispatcher will not know anything. What if you could use a unique id as aframeworkId plus an increment, so the scheduler can detect all frameworks even the ones coming from multiple contexts?